### ISSUE TITLE
Graphics VRAM increase

### ISSUE THREAD

**Issue by startergo** [royalgraphx/LegacyOSXKVM#6]
I installed the SL on QEMU in Sonoma. Everything runs good, but 3 MB Vram memory is not enough. Is there a way to add more Vram? I tried adding this kext https://github.com/ivanagui2/VMQemuVGA/tree/master and installed Vmware Darwin.iso, but I don't see a change.
Also is there a way to convert the Libvirt xml to a UTM template?

**Comment by royalgraphx** [comment]
Due to many limitations that I don't really want to talk about because its super long to explain but virtual displays are simply just not the solution, and there is no way to get any better graphics acceleration on virtual displays, because there is no real Graphics driver, just the display and its using VGA as the driver... the only way is to look for AMD dedicated GPUs that work under Snow Leopard, and do GPU passthrough, that's literally the only way, and it's what we already have to do for latest macOS, and legacy OS X is no different.

TL;DR, buy a supported AMD dGPU and do GPU passthrough, Virtual Displays are VGA, that's all

**Comment by startergo** [comment]
I managed to run it in the UTM with 7mb instead of 3MB Vram and it is much better now. For some reason I can't get internet working maybe you can shed some light?
https://github.com/utmapp/UTM/issues/6338

**Comment by royalgraphx** [comment]
try not to be mixing projects together, UTM is simply a front-end for QEMU, but uses its own forks for support on iOS/macOS. Every attempt i've done with QEMU/UTM VGA leads to 3,5,7MB VRAM, it's just been random, but my point is still the same, it will not increase and even if you use VMware to get 128mb or wtv, you still lack hardware to encode/decode, it will always be just a virtual display, and it's only that OS X Snow Leopard and older are so light, they don't rely on Graphics Acceleration like macOS Ventura or Sonoma do, as in, the entire UI and animations for everything.

if the internet is not working, simply emulate the same ethernet controller that I do on LegacyOSXKVM.

Wi-Fi does not work on VM's... it's host internet connection shared to the guest using virtual ethernet, simply use the correct virtual ethernet controller that has drivers in the appropriate Guest OS you are attempting to emulate/virtualize

**Comment by startergo** [comment]
>  and it's only that OS X Snow Leopard and older are so light, they don't rely on Graphics Acceleration like macOS Ventura or Sonoma do, as in, the entire UI and animations for everything.


Correct. 7MB is pretty good for SL I don't need more.



> if the internet is not working, simply emulate the same ethernet controller that I do on LegacyOSXKVM.
> 
> Wi-Fi does not work on VM's... it's host internet connection shared to the guest using virtual ethernet, simply use the correct virtual ethernet controller that has drivers in the appropriate Guest OS you are attempting to emulate/virtualize

I tried emulating, sharing, bridging all adapters. Only the 1000e sharing and bridging works. I can do remote control through ARD in bridge mode, but no internet connection. And it works in QEMU directly with your setup. I don't get what gives.

**Comment by royalgraphx** [comment]
known issues tbh, nothing new here

if you're experiencing an issue with DNS resolution on Snow Leopard refer to the text below



sudo launchctl unload /System/Library/LaunchDaemons/com.apple.mDNSResponder.plist

sudo launchctl load -F /System/Library/LaunchDaemons/com.apple.mDNSResponder.plist

ping 8.8.8.8

**Comment by startergo** [comment]
> known issues tbh, nothing new here
> 
> if you're experiencing an issue with DNS resolution on Snow Leopard refer to the text below
> 
> sudo launchctl unload /System/Library/LaunchDaemons/com.apple.mDNSResponder.plist
> 
> sudo launchctl load -F /System/Library/LaunchDaemons/com.apple.mDNSResponder.plist
> 
> ping 8.8.8.8

You are the man! Thanks!

**Comment by startergo** [comment]
As a proof of concept:
![Screen Shot 2024-06-11 at 8 47 45 PM](https://github.com/royalgraphx/LegacyOSXKVM/assets/7897244/c67a672f-1913-4226-b0f8-14383c3dd536)
